src/Gir/GenerationOptions.cs

```diff
@@ -34,26 +35,29 @@ public sealed class GenerationOptions
 
 		#endregion
 
-		IEnumerable<Repository> AllRepositories { get; }
+		IEnumerable<Repository> ResolvedRepositories { get; }
 		Repository Repository { get; }
 		List<IGeneratable> allGeneratables = new List<IGeneratable> ();
 
-		public GenerationOptions (string dir, IEnumerable<Repository> allRepos, Repository repo, ToggleOptions options = null)
+		public GenerationOptions (string dir, IEnumerable<Repository> resolvedRepos, Repository repo, ToggleOptions options = null)
 		{
 			// Set options to default if none
 			Options = options ?? new ToggleOptions ();
 
 			DirectoryPath = dir;
-			AllRepositories = allRepos;
+			ResolvedRepositories = resolvedRepos;
 			Repository = repo;
 			allGeneratables.AddRange (Repository.GetGeneratables ());
 
 			// This may contain multiple libraries, so get the first one. Also, hack a string.Empty for xlib.
 			LibraryName = repo.Namespace.SharedLibrary?.Split (',') [0] ?? "";
 
-			SymbolTable = new SymbolTable (Statistics, options.Win64Longs);
-			SymbolTable.AddTypes (allRepos.SelectMany (x => x.GetSymbols ()));
-			SymbolTable.ProcessAliases ();
+			SymbolTable = new SymbolTable(Statistics, options.Win64Longs);
+			SymbolTable.AddTypes (repo.GetSymbols());
+			foreach (var repository in resolvedRepos) {
+				SymbolTable.AddTypes (repository.GetSymbols(), repository);
```

Comment:
This includes the main repository so the mainrepository is in the symbol table twice: Once without and once with namespace prefix. This is because arrays seem to behave weird. There are array types inside GLib which have GLib.ByteArray as type where it should have been just ByteArray. That's why I added the repository twice.